diff --git a/build/config/android/config.gni b/build/config/android/config.gni
index f02a355..427c444 100644
--- a/build/config/android/config.gni
+++ b/build/config/android/config.gni
@@ -15,6 +15,8 @@ if (is_android) {
     android_sdk_root = default_android_sdk_root
     android_sdk_version = default_android_sdk_version
     android_sdk_build_tools_version = default_android_sdk_build_tools_version
+    android_ndk_root = "//flutter/third_party/android_tools/ndk"
+    android_toolchain_clang_version = "17"
 
     # Unused. Required for GN files maintained in other buildroots.
     enable_java_templates = false
@@ -53,7 +55,6 @@ if (is_android) {
   android_sdk = "${android_sdk_root}/platforms/android-${android_sdk_version}"
 
   # Path to the Android NDK and SDK.
-  android_ndk_root = "//flutter/third_party/android_tools/ndk"
   android_ndk_include_dir = "$android_ndk_root/usr/include"
 
   android_sdk = "${android_sdk_root}/platforms/android-${android_sdk_version}"
diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 61e54bb..805dd33 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -560,7 +560,9 @@ config("runtime_library") {
       "m",
     ]
 
-    lib_dirs += [ "${android_toolchain_root}/lib/clang/17/lib/linux/" ]
+    lib_dirs += [
+      "${android_toolchain_root}/lib/clang/${android_toolchain_clang_version}/lib/linux/",
+    ]
     current_android_cpu = current_cpu
     if (current_cpu == "arm64") {
       current_android_cpu = "aarch64"
diff --git a/build/linux/sysroot_scripts/install-sysroot.py b/build/linux/sysroot_scripts/install-sysroot.py
index d3e7664..f03fc9d 100755
--- a/build/linux/sysroot_scripts/install-sysroot.py
+++ b/build/linux/sysroot_scripts/install-sysroot.py
@@ -151,7 +151,14 @@ def InstallSysroot(target_platform, target_arch):
   if sha1sum != tarball_sha1sum:
     raise Error('Tarball sha1sum is wrong.'
                 'Expected %s, actual: %s' % (tarball_sha1sum, sha1sum))
-  subprocess.check_call(['tar', 'xf', tarball, '-C', sysroot])
+  subprocess.check_call([
+      'tar',
+      '--no-same-owner',
+      '-xf',
+      tarball,
+      '-C',
+      sysroot,
+  ])
   os.remove(tarball)
 
   with open(stamp, 'w') as s:
